db/migrate/20231103061213_add_terms_of_service_accepted_at_to_spree_users.rb

```diff
@@ -3,7 +3,18 @@ def up
     add_column :spree_users, :terms_of_service_accepted_at, :datetime
 
     if Spree::Config.enterprises_require_tos == true
-      Spree::User.update_all(terms_of_service_accepted_at: Time.zone.now)
+      # Update only user who are owners of at least 1 enterprise
+      ActiveRecord::Base.sanitize_sql(["
+         UPDATE spree_users
+         SET terms_of_service_accepted_at = :time
+         WHERE spree_users.id IN(
+           SELECT id FROM spree_users WHERE EXISTS (
+             SELECT 1 FROM enterprises WHERE spree_users.id = enterprises.owner_id
+           )
+        )".squish,
+        time: Time.zone.now
+      ])
+      ActiveRecord::Base.connection.execute(sql)
```

Comment:
I just had a look at who can access the admin pages and there are lots more cases. You may not have an enterprise but you may have a group or you may manage an enterprise.

https://github.com/openfoodfoundation/openfoodnetwork/blob/907d559eeffdd1c1063645f76f17d9ce16ffbc3f/app/models/spree/ability.rb#L49-L54

Technically it's wrong to say that shoppers accepted the ToS but they have to accept the terms when the sign up for an enterprise. So I reckon that your previous solution was actually okay.

It does make me realise though that we don't remember the checkbox in the registration process though. We need to set `terms_of_service_accepted_at` when the first enterprise is created through the registration pages (API calls, I guess).

Reply:
I'll revert this then , anyway I had fun writing some SQL.

> It does make me realise though that we don't remember the checkbox in the registration process though. We need to set terms_of_service_accepted_at when the first enterprise is created through the registration pages (API calls, I guess).

This is tracked here : https://github.com/openfoodfoundation/openfoodnetwork/issues/11796